app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -116,13 +117,21 @@ function RawScatterPlot({
     return scatter
   }
 
+  /** redraw the plot when editedCustomColors changes */
+  useEffect(() => {
+    if (editedCustomColors && Object.keys(editedCustomColors).length > 0) {
+      const plotlyTraces = document.getElementById(graphElementId).data
+      Plotly.react(graphElementId, plotlyTraces, scatterData.layout)
+    }
+  }, [editedCustomColors])
+
   /** Save any changes to the legend colors */
-  async function saveCustomColors(newColors) {
+  const saveCustomColors = async newColors => {
```

Comment:
Blocker: SCP JS uses classic function signatures for these things, which helps look up functions when one's IDE is having issues.  If we change that convention, I'd advocate doing it as a holistic sweep, for consistency.

```suggestion
  async function saveCustomColors(newColors) {
```

Reply:
oops yes will flip that back, force of habit haha